provider.go

```diff
@@ -85,6 +88,25 @@ func newAzureProvider(settings *clientSettings) (AzureProvider, error) {
 		userAgent = useragent.String()
 	}
 
+	// Sets a unique ID in the user-agent
+	// Normal user-agent looks like this:
+	//
+	// Vault/1.6.0 (+https://www.vaultproject.io/; azure-secrets; go1.15.7)
+	//
+	// Here we append a unique code if it's an enterprise version, where
+	// VersionMetadata will contain a non-empty string like "ent" or "prem".
+	// Otherwise use the default identifier for OSS Vault. The end result looks
+	// like so:
+	//
+	// Vault/1.6.0 (+https://www.vaultproject.io/; azure-secrets; go1.15.7; b2c13ec1-60e8-4733-9a76-88dbb2ce2471)
+	vaultIDString := "; 15cd22ce-24af-43a4-aa83-4c1a36a4b177)"
+	ver := version.GetVersion()
```

Comment:
This might be not be accurate if the plugin is compiled and ran externally on a Vault Enterprise installation since this will always be pointing to the OSS SDK version value.

Reply:
Unfortunately I wasn't able to find a way of getting any metadata or otherwise information on being Enterprise or not when compiling the plugin separately. Do you have any suggestions, or a better way of handling it?